cleverhans/attacks.py

```diff
@@ -47,7 +57,7 @@ def __init__(self, model, back='tf', sess=None, dtypestr='float32'):
 
     # Prepare attributes
     self.model = model
-    self.back = back
+    self.back = 'tf'
```

Comment:
Do we still need self.back at all?

Reply:
If we need self.back for backwards compatibility purposes, maybe we should make it a python property ( https://www.programiz.com/python-programming/property ) that issues a deprecation warning.